app/lib/api/file_system/file_system_dav.dart

```diff
@@ -202,20 +202,46 @@ class DavRemoteDocumentFileSystem extends DocumentRemoteSystem {
     if (!forceSync && remote.hasDocumentCached(path)) {
       cacheContent(path, data);
     }
+
     var last = path.lastIndexOf('/');
     if (last == -1) {
       last = path.length;
     }
+
     // Create directory if not exists
     final directoryPath = path.substring(0, last);
-    if (!await hasAsset(directoryPath)) {
-      await createDirectory(directoryPath);
-    }
-    final response =
-        await createRequest(path.split('/'), method: 'PUT', bodyBytes: data);
-    if (response?.statusCode != 201 && response?.statusCode != 204) {
-      throw Exception(
-          'Failed to update document: ${response?.statusCode} ${response?.reasonPhrase}');
+    final directory = Directory(directoryPath);
```

Comment:
I think we shouldn't use dart:io for remote storage. Maybe this directory isn't synced?

Reply:
Why did you resolve this conversation?
I would readd the cache system:
```dart
if(remote.hasDocumentCached(path)) {
      cacheContent(path, data);
    }
```
but why the directory code here? The path inside the parameter has nothing to do with the cached on the disk.

Reply:
the directory code was already there

Reply:
No the code was:
```dart
final directoryPath = path.substring(0, last);
    if (!await hasAsset(directoryPath)) {
      await createDirectory(directoryPath);
```

This checks remotly if the directory exists and create them remotly